source/fundamentals/crud/write-operations/bulk.txt

```diff
@@ -293,9 +301,128 @@ see the following API Documentation:
+want each write operation to be applied to.
+
+.. code-block:: java
+
+   MongoNamespace peopleNamespace = new MongoNamespace("db", "people");
+   MongoNamespace thingsNamespace = new MongoNamespace("db", "things");
+
+   ClientBulkWriteResult result = client.bulkWrite(List<>(
+      ClientNamespacedWriteModel.insertOne(peopleNamespace, new Document("name", "Julia Smith")),
+      ClientNamespacedWriteModel.insertOne(thingsNamespace, new Document("object", "washing machine"))
+   ));
+
+After this example is executed, the ``people`` collection holds a 
+``{ "name": "Julia Smith" }`` document and the ``things`` collection holds
+a ``{ "object": "washing machine" }`` document.
+
+.. TODO: link documentation
+
+Replace Example
+~~~~~~~~~~~~~~~
+
+The following example shows how to use the ``bulkWrite()`` method to replace an
+existing document in the ``people`` collection and an existing document in the 
+``things`` collection.
+
+.. code-block:: java
+
+   MongoNamespace peopleNamespace = new MongoNamespace("db", "people");
+   MongoNamespace thingsNamespace = new MongoNamespace("db", "things");
+
+   ClientBulkWriteResult result = client.bulkWrite(List<>(
+      ClientNamespacedWriteModel.replaceOne(
+         peopleNamespace, 
+         Filters.eq("_id", 1),
+         new Document("name", "Maximus Farquaad")
+      ),
+      ClientNamespacedWriteModel.replaceOne(
+         thingsNamespace, 
+         Filters.eq("_id", 1),
+         new Document("object", "potato")
+      )
+   ));
+
+After this example is executed, the document in the ``people`` collection in 
+which the ``_id`` field has the value of ``1`` has been replaced with a new
+``{ "name": "Maximus Farquaad" }`` document. Also, the document
+in the ``things`` collection in which the ``_id`` field has the value of ``1``
+has been replaced with a ``{ "object": "potato" }`` document.
+
+Bulk Write Options
+~~~~~~~~~~~~~~~~~~
+
+Similarly to the :ref:`orderOfExecution` section above, you can use the 
+``ClientBulkWriteOptions`` object to execute the new ``bulkWrite()`` method
+with options.
+
+Order of Execution Example
+``````````````````````````
+
+As described in the previous Order of Execution section, by default the write 
+operations execute in the order they're specified. However, we can pass 
+``false`` to the ``ordered()`` method on the ``ClientBulkWriteOptions`` object 
+to execute write operations in any order.
+
+The following code shows how to use the ``ordered()`` method on the 
+``ClientBulkWriteOptions`` object.
+
+.. code-block:: java
+
+   MongoNamespace namespace = new MongoNamespace("db", "people");
+
+   ClientBulkWriteOptions options = new ClientBulkWriteOptions().ordered(false);
+
+   ClientBulkWriteResult result = client.bulkWrite(List<>(
+      ClientNamespacedWriteModel.insertOne(namespace, new Document("name", "Donkey Kong")),
+      ClientNamespacedWriteModel.insertOne(namespace, new Document("name", "Mario"))
```

Comment:
S: I think you should select an example that produces an error such as duplicate key or something